src/graph/offline-query-lowest-common-ancestor.hpp

```diff
@@ -9,56 +10,61 @@
 namespace luz {
 
   template< typename cost_type >
-  class OfflineLCAQuery: Graph< cost_type > {
-    using Graph< cost_type >::g;
+  class OfflineLCAQuery {
+    usize g_size_;
+    Graph< cost_type > g_;
 
-    std::vector< std::vector< std::pair< usize, usize > > > qs;
+    usize query_count_;
+    std::vector< std::vector< std::pair< usize, usize > > > qs_;
 
-    DisjointSetUnion dsu;
-    std::vector< bool > visited;
-    std::vector< usize > ancestors;
+    std::vector< std::unordered_map< usize, usize > > to_qi;
+
+    DisjointSetUnion dsu_;
+    std::vector< bool > visited_;
+    std::vector< usize > ancestors_;
+
+    std::vector< usize > ret;
 
```

Comment:
```suggestion

    void bound_check(usize v) const {
      assert(v < n_);
    }
```

をして `build(root)` とか `lca(u, v)` とかをよしなに